imgui-SFML.cpp

```diff
@@ -529,45 +529,39 @@ void Render() {
 }
 
 void Shutdown(const sf::Window& window) {
+    if (s_windowContexts.find(window.getSystemHandle()) == s_windowContexts.end()) {
+        assert(false &&
+                "Window wasn't inited properly: forgot to call ImGui::SFML::Init(window)?");
+    }
+
+    // check if the current window's context is the one we want to remove
+    const bool isCurrentWindowCtx =
+        window.getSystemHandle() == s_currWindowCtx->window->getSystemHandle();
+
+    // remove window's context  
+    delete s_windowContexts[window.getSystemHandle()];
+    s_windowContexts.erase(window.getSystemHandle());
+
     // set current context to some window for convenience if needed
-    if (window.getSystemHandle() == s_currWindowCtx->window->getSystemHandle()) {
```

Comment:
At this point:
```c++
    printf("%d %d\n", window.getSystemHandle(), s_currWindowCtx->window->getSystemHandle());
```
on Linux this prints `0 0`.

That is, after a window is closed, its handle changes to 0.
So basically previously it kinda worked by accident, the condition is basically this: if the handle of the passed window is closed and the handle of the current window is closed, assume they're the same window.
Because the usage of `map` assumes the `SystemHandle` will stay constant and valid, the new logic is not equivalent and causes a crash.